admin/class-admin-apple-bulk-export-page.php

```diff
@@ -96,7 +96,27 @@ public function ajax_push_post() {
 		// Sanitize input data
 		$id = absint( $_GET['id'] );
 
-		// TODO: Move push action to shared
+		// Ensure the post exists and that it's published
+		$post = get_post( $id );
+		if ( empty( $post ) ) {
+			echo json_encode( array(
+				'success' => false,
+				'error'   => __( 'This post no longer exists.', 'apple-news' ),
+			) );
+			wp_die();
+		}
+
+		if ( 'publish' != $post->post_status ) {
+			echo json_encode( array(
+				'success' => false,
+				'error'   => sprintf(
+					__( 'Article %s is not published and cannot be pushed to Apple News.', 'apple-news' ),
+					$id
+				),
+			) );
+			wp_die();
```

Comment:
Also not a crucial change, but 110-117 could also use `wp_send_json_error()` to save some lines.